doc/languages-frameworks/rust.md

```diff
@@ -57,6 +57,83 @@ checksum can be then take from the failed build.
 To install crates with nix there is also an experimental project called
 [nixcrates](https://github.com/fractalide/nixcrates).
 
+## Compiling Rust crates using Nix instead of Cargo
+
+When run, `cargo build` produces a file called `Cargo.lock`,
+containing pinned versions of all dependencies. Nixpkgs contains a
+tool called `carnix` (`nix-env -iA nixos.carnix`), which can be used
+to turn a `Cargo.lock` into a Nix expression.
+
+That Nix expression calls `rustc` directly (hence bypassing Cargo),
+and can be used to compile a crate and all its dependencies. Here is
+an example for a minimal `hello` crate:
+
+
+    $ cargo new hello
+    $ cd hello
+    $ cargo build
+     Compiling hello v0.1.0 (file:///tmp/hello)
+      Finished dev [unoptimized + debuginfo] target(s) in 0.20 secs
+    $ carnix -o hello.nix --src ./. Cargo.lock
+    $ nix-build hello.nix
+
+Now, the file produced by the call to `carnix`, called `hello.nix`, looks like:
+
+```
+with import <nixpkgs> {};
+let release = true;
+    verbose = true;
+    hello_0_1_0_ = { dependencies?[], buildDependencies?[], features?[] }: buildRustCrate {
+      crateName = "hello";
+      version = "0.1.0";
+      src = ./.;
+      inherit dependencies buildDependencies features release verbose;
+    };
+
+in
+rec {
+  hello_0_1_0 = hello_0_1_0_ {};
+}
+```
+
+In particular, note that the argument given as `--src` is copied
+verbatim to the source.  If we look at a more complicated
+dependencies, for instance by adding a single line `libc="*"` to our
+`Cargo.toml`, we get the following `hello.nix`:
```

Comment:
When I add a single dependency like you describe:

```
[package]
name = "hello"
version = "0.1.0"
authors = ["Joerg Thalheim <joerg@thalheim.io>"]
[dependencies]
libc = "*"
```

```
[root]
name = "hello"
version = "0.1.0"
dependencies = [
 "libc 0.2.33 (registry+https://github.com/rust-lang/crates.io-index)",
]
[[package]]
name = "libc"
version = "0.2.33"
source = "registry+https://github.com/rust-lang/crates.io-index"
[metadata]
"checksum libc 0.2.33 (registry+https://github.com/rust-lang/crates.io-index)" = "5ba3df4dcb460b9dfbd070d41c94c19209620c191b0340b929ce748a2bcd42d2"
```

`carnix` returns the following:

```console
$ carnix -o hello.nix --src ./. Cargo.lock
Prefetching libc-0.2.33
nix-prefetch-url returned:
unpacking...
path is '/nix/store/pmmp53ac3zcfwjv4vr9ap49di86j3q73-libc-0.2.33'
```

without writing anything to `hello.nix`. I works though if there are no dependencies in `Cargo.toml`:

```with import <nixpkgs> {};
let release = true;
    verbose = true;
    hello_0_1_0_ = { dependencies?[], buildDependencies?[], features?[] }: buildRustCrate {
      crateName = "hello";
      version = "0.1.0";
      src = ./.;
      inherit dependencies buildDependencies features release verbose;
    };
in
rec {
  hello_0_1_0 = hello_0_1_0_ {};
}
```

Reply:
@Mic92  Last time I tried to run this code without dependencies in toml file it didn't work.

Reply:
Have you run `cargo build` after adding the new dependency?

Reply:
@P-E-Meunier yes, you can see how libc was added to `Cargo.lock`.

Reply:
Alright, so it seems the apostrophes used delimit the path have several variants, and the regular expression used to parsed that output was only handling one.
The one I get here is ‘, whereas yours seems to be a '.

I've uploaded carnix 0.4.4 to crates.io now to correct for that.